distributed/scheduler.py

```diff
@@ -3013,6 +3017,24 @@ def get_task_stream(self, comm=None, start=None, stop=None, count=None):
         ts = [p for p in self.plugins if isinstance(p, TaskStreamPlugin)][0]
         return ts.collect(start=start, stop=stop, count=count)
 
+    @gen.coroutine
+    def register_init_func(self, comm, function=None):
+        """ Registers a preload function, and call it on every worker """
+        if function is None:
+            raise gen.Return({})
+
+        self.init_functions.append(loads(function))
```

Comment:
If possible, I think it would be better to never call `pickle.loads` in the scheduler.  The scheduler tries to never actually touch real Python code.  Instead, I would store the serialized bytes in the scheduler and then pass those on directly.  

I'll also offer a suggestion in a comment next to the client code.

Reply:
Done.